src/parser/paladin/protection/modules/talents/Seraphim.tsx

```diff
@@ -0,0 +1,99 @@
+import React from 'react';
+import SPELLS from 'common/SPELLS';
+import Analyzer, { Options, SELECTED_PLAYER } from 'parser/core/Analyzer';
+import Events, { CastEvent } from 'parser/core/Events';
+import StatTracker from 'parser/shared/modules/StatTracker';
+import Statistic from 'interface/statistics/Statistic';
+import STATISTIC_ORDER from 'interface/others/STATISTIC_ORDER';
+import STATISTIC_CATEGORY from 'interface/others/STATISTIC_CATEGORY';
+import { formatNumber, formatPercentage } from 'common/format';
+import UptimeIcon from 'interface/icons/Uptime';
+import CriticalStrikeIcon from 'interface/icons/CriticalStrike';
+import VersatilityIcon from 'interface/icons/Versatility';
+import HasteIcon from 'interface/icons/Haste';
+import MasteryIcon from 'interface/icons/Mastery';
+import BoringSpellValueText from 'interface/statistics/components/BoringSpellValueText';
+
+const STAT_MODIFIER = 0.08;
+
+type BonusStats = {haste: number, crit: number, versatility: number, mastery: number};
+
+/**
+ * Consumes 3 Holy Power to grant 8% Haste, Crit, Versatility, and Mastery for 15 seconds.
+ */
+class Seraphim extends Analyzer {
+  static dependencies = {
+    statTracker: StatTracker,
+  };
+
+  protected statTracker!: StatTracker;
+
+  bonusStatGains: BonusStats[] = [];
+
+
+  constructor(options: Options) {
+    super(options);
+    this.active = this.selectedCombatant.hasTalent(SPELLS.SERAPHIM_TALENT.id);
+    if (!this.active) {
+      return;
+    }
+    this.addEventListener(Events.cast.by(SELECTED_PLAYER).spell(SPELLS.SERAPHIM_TALENT), this.onCast);
+    const localStatTracker = (options.statTracker as StatTracker);
+    const bonusStats = this.getBonusStats(localStatTracker);
+    localStatTracker.add(SPELLS.SERAPHIM_TALENT.id, bonusStats);
```

Comment:
sera also multiplies stats gained from other buffs (e.g. trinkets), which means that this value is going to be an underestimate.

importantly: this also means that haste will be wrong when sera + another haste buff is active

Reply:
Do we have support for calculating the overall stat gain with other buffs taken into account through `StatTracker`? A quick look through the code indicates that `StatTracker` just applies stat rating changes and stat rating multipliers iteratively as buffs are applied and fall off which indicates that buffs are treated as snapshots rather than dynamic values.

Reply:
So i did some testing on Seraphim on live (with no trinkets / azerite /essences so no procs). Here's the before and after:

## Before
![Capture](https://user-images.githubusercontent.com/4909458/97597974-cc899400-19dc-11eb-85f1-d3625c5d0e25.PNG)
## After

![image](https://user-images.githubusercontent.com/4909458/97597932-c398c280-19dc-11eb-9ef5-6f82aa8ed6d4.png)

Notably, these are all exactly 8% stat gain---not 8% rating gain---with the exception of Haste, which uses the 8% rating + 8% stat gain rule (see [here](https://github.com/WoWAnalyzer/WoWAnalyzer/blob/shadowlands/src/parser/shared/modules/Haste.js#L223-L225)). % Stat gain ignores DRs, which means that this will always be 8% (except haste, because reasons).

That means that we actually need some new machinery in `StatTracker` to handle Vers/Crit/Mastery. However, the `Haste` module already works. Since we *mostly* need that to be correct, we could merge this with Sera haste added to the Haste module and the others just showing on the tooltip as `uptime * 8%`.

Reply:
Regarding the changes needed for `StatTracker` to handle the non-haste stats: It looks like StatTracker currently supports tracking percentage-based buffs, it just lacks a mechanism for adding new buffs to the list of currently tracked buffs. It should be fairly simple for me to add what's needed in this CR.

Otherwise, I've created a prot `Haste` module following the line of the ret `Haste ` module and registered the Seraphim talent with that.

Reply:
Resolved in latest commit.